src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -202,7 +218,7 @@ private Object createObjectWithIgnoringProblems(String className, Set<String> se
      * @return a string which is obtained by joining package names with a class name.
      */
     private static String joinPackageNamesWithClassName(String className, Set<String> packages) {
-        return packages.stream().filter(name -> name != null)
+        return packages.stream().filter(Objects::nonNull)
```

Comment:
> IMO, it doesn't make sense to have a filter for non-null check here.

I don't know the original reason. Could be that they didn't want to modify the variable coming in the constructor?
I am now curious if we have any current tests using a `null` package.

@romani Do you know the reason for this `null` check?
Can we remove it and assert the list must never have a `null` item?